pkg/didcomm/transport/ws/inbound.go

```diff
@@ -73,77 +75,22 @@ func (i *Inbound) Endpoint() string {
 	return i.externalAddr
 }
 
-func newInboundHandler(prov transport.InboundProvider) (http.Handler, error) {
-	if prov == nil || prov.InboundMessageHandler() == nil {
-		logger.Errorf("Error creating a new inbound handler: message handler function is nil")
-		return nil, errors.New("creation of inbound handler failed")
-	}
-
-	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		processRequest(w, r, prov)
-	}), nil
-}
-
-func processRequest(w http.ResponseWriter, r *http.Request, prov transport.InboundProvider) {
-	c, cleanup, err := upgradeConnection(w, r)
+func (i *Inbound) processRequest(w http.ResponseWriter, r *http.Request) {
+	c, err := upgradeConnection(w, r)
 	if err != nil {
 		logger.Errorf("failed to upgrade the connection : %v", err)
 		return
 	}
 
-	defer cleanup()
-
-	for {
-		_, message, err := c.Read(context.Background())
-		if err != nil {
-			if websocket.CloseStatus(err) != websocket.StatusNormalClosure {
-				logger.Errorf("Error reading request message: %v", err)
-			}
-
-			break
-		}
-
-		unpackMsg, err := prov.Packager().UnpackMessage(message)
-		if err != nil {
-			logger.Errorf("failed to unpack msg: %v", err)
-
-			err = c.Write(context.Background(), websocket.MessageText, []byte(processFailureErrMsg))
-			if err != nil {
-				logger.Errorf("error writing the message: %v", err)
-			}
-
-			continue
-		}
-
-		messageHandler := prov.InboundMessageHandler()
-
-		resp := ""
-
-		err = messageHandler(unpackMsg.Message)
-		if err != nil {
-			logger.Errorf("incoming msg processing failed: %v", err)
-
-			resp = processFailureErrMsg
-		}
-
-		err = c.Write(context.Background(), websocket.MessageText, []byte(resp))
-		if err != nil {
-			logger.Errorf("error writing the message: %v", err)
-		}
-	}
+	i.pool.listener(c)
 }
 
-func upgradeConnection(w http.ResponseWriter, r *http.Request) (*websocket.Conn, func(), error) {
+func upgradeConnection(w http.ResponseWriter, r *http.Request) (*websocket.Conn, error) {
```

Comment:
Can you add // +build !js

Reply:
To resolve https://github.com/hyperledger/aries-framework-go/issues/879, adding this tag will not suffice. Need to update the dependencies. I'll add it as a separate PR (this PR is already > 500 lines).